src/main/java/ro/vladutit/Don/t/forget/v2/controller/CategoryController.java

```diff
@@ -1,22 +1,34 @@
 package ro.vladutit.Don.t.forget.v2.controller;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 import ro.vladutit.Don.t.forget.v2.model.Category;
+import ro.vladutit.Don.t.forget.v2.model.CustomUserDetails;
 import ro.vladutit.Don.t.forget.v2.model.Icon;
+import ro.vladutit.Don.t.forget.v2.model.User;
 import ro.vladutit.Don.t.forget.v2.service.CategoryService;
+import ro.vladutit.Don.t.forget.v2.service.CustomUserDetailsService;
+import ro.vladutit.Don.t.forget.v2.service.UserService;
 
 @Controller
 public class CategoryController implements WebMvcConfigurer {
 
     @Autowired
     private CategoryService categoryService;
 
+    @Autowired
+    private UserService userService;
+
     // Save category
     @PostMapping("/category/save")
-    public String addNew(Category category) {
+    public String addNew(Category category, @AuthenticationPrincipal CustomUserDetails userDetails) {
+        String userEmail = userDetails.getUsername();
+        User user = userService.getByEmail(userEmail);
+        category.setUser(user);
```

Comment:
can you explain why do we need this? I don't have a clear enough picture for this

Reply:
Each **category** will now save the _id_ of the **user** who created that **category**. I could probably do this addition from the _frontend_, but for _security_, I think it's better to have this association in the _backend_. In these lines, with the help of the **AuthenticationPrincipal** _annotation_, I detect the user who is logged through the **CustomUserDetails** _class_. In this class, the login information (username and password) are retained, and to get to the _id_, I look for the _username_ (which is actually the _email_) in the **User** _table_. I find the respective **user** that I save later in the **categories** _table_. I could probably save the _username_ directly in the **category** _table_, but I think it's more **private** and **safe** to save an _id_, instead of exact information such as an _email_ address.